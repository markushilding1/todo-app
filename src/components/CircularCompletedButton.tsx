import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import colors from '../constants/colors';

interface CircularCompletedButtonProps {
  onPress: () => void;
  disabled?: boolean;
  completed?: boolean;
}

const CircularCompletedButton = ({
  onPress,
  disabled,
  completed,
}: CircularCompletedButtonProps) => {
  return (
    <TouchableOpacity
      testID="circularCompletedButton"
      onPress={onPress}
      disabled={disabled}
      style={[styles.button, completed && styles.completed]}
    />
  );
};

const styles = StyleSheet.create({
  button: {
    width: 24,
    height: 24,
    borderRadius: 15,
    borderWidth: 1,
    borderColor: colors.placeholderText,
  },
  completed: {
    backgroundColor: colors.primary,
    borderWidth: 0,
  },
});

export default CircularCompletedButton;
