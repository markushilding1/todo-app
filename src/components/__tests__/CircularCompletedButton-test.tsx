import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react-native';

import CircularCompletedButton from '../CircularCompletedButton';

describe('button', () => {
  describe('render', () => {
    const mockFunction = jest.fn();

    const button = renderer.create(
      <CircularCompletedButton
        onPress={mockFunction}
        completed
        disabled={false}
      />,
    );

    it(`renders correctly`, () => {
      expect(button.toJSON()).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it(`calls function prop "onPress"`, async () => {
      const mockFunction = jest.fn();
      render(<CircularCompletedButton onPress={mockFunction} />);
      fireEvent.press(screen.getByTestId('circularCompletedButton'));
      expect(mockFunction).toHaveBeenCalledTimes(1);
    });

    it(`does not call function prop "onPress"`, () => {
      const mockFunction = jest.fn();
      render(<CircularCompletedButton onPress={mockFunction} disabled />);
      fireEvent.press(screen.getByTestId('circularCompletedButton'));
      expect(mockFunction).toHaveBeenCalledTimes(0);
    });
  });
});
