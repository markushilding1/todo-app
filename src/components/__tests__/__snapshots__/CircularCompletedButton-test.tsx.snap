// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#206795",
      "borderColor": "#95ACAE",
      "borderRadius": 15,
      "borderWidth": 0,
      "height": 24,
      "opacity": 1,
      "width": 24,
    }
  }
  testID="circularCompletedButton"
/>
`;
